src/frontend/src/optimizer/rule/batch/batch_iceberg_predicate_pushdown.rs

```diff
@@ -84,8 +84,8 @@ fn rw_literal_to_iceberg_datum(literal: &Literal) -> Option<IcebergDatum> {
             };
             Some(datum)
         }
-        ScalarImpl::Timestamp(t) => Some(IcebergDatum::timestamp_nanos(
-            t.0.and_utc().timestamp_nanos_opt()?,
+        ScalarImpl::Timestamp(t) => Some(IcebergDatum::timestamp_micros(
+            t.0.and_utc().timestamp_micros(),
```

Comment:
Are we fixing a different bug here? And do we have a test covering this branch 🤪

Reply:
Just found that timestamp is microsecond precision for both risingwave and iceberg 😆

Reply:
`date <= DATE '2025-02-10' - interval '5' day`

So the right side changes from `timestamp_nanos` to `timestamp_micros`.
And then will we pushdown or not predicate for it after the PR?

I don't fully get the reason of the type change here. 🤔

Reply:
> date <= DATE '2025-02-10' - interval '5' day
We won't push down this predicate because the lhs with data type `date` and the rhs is `timestamp`.
We can push down the predicate if we change it to `date <= (DATE '2025-02-10' - interval '5' day)::date`

Reply:
> Just found that timestamp is microsecond precision for both risingwave and iceberg 😆

I found iceberg has a `timestamp_ns` type. And we've supported to convert it to RisingWave `timestamp` in #19221

So it seems using either one is ok here...? 🤔

Reply:
We can support timestamp_ns after this PR https://github.com/risingwavelabs/risingwave/pulls?q=is%3Apr+is%3Aopen+timestamp+ns